### PR TITLE
DHFPROD-1453: hubSaveIndexes gradle task to save indexes

### PR DESCRIPTION
**Pull request by rahulvudutala**
This PR has commits regarding

* gradle task for saving indexes from entities on file system
* Changes in saveDbIndexes tests in marklogic-data-hub
* ml-data-hub unit test for hubSaveIndexes task

### CODE REVIEW

File: marklogic-data-hub/build.gradle [hunk 1]
```diff
@@ -32,6 +32,7 @@ dependencies {
     compile 'com.marklogic:marklogic-data-movement-components:1.0'
     compile 'commons-io:commons-io:2.4'
     compile 'org.apache.commons:commons-text:1.1'
+    compile 'org.json:json:20180813'
```
Comment (grechaw) [inline marklogic-data-hub/build.gradle]: you should not need to include this.  I'll find where you imported it and suggest an alternative

File: marklogic-data-hub/src/main/java/com/marklogic/hub/impl/EntityManagerImpl.java [hunk 2]
```diff
@@ -128,15 +136,23 @@ private boolean deployQueryOptions(DatabaseClient client, String filename) {
...
             if (!dir.toFile().exists()) {
                 dir.toFile().mkdirs();
             }
-            File finalFile = Paths.get(dir.toString(), HubConfig.FINAL_ENTITY_DATABASE_FILE).toFile();
-            File stagingFile = Paths.get(dir.toString(), HubConfig.STAGING_ENTITY_DATABASE_FILE).toFile();
 
-            long lastModified = Math.max(finalFile.lastModified(), stagingFile.lastModified());
-            List<JsonNode> entities = getModifiedRawEntities(lastModified);
+            List<JsonNode> entities = getAllEntities();
+            List<String> databaseEntities = getEntityNamesInDatabase();
```
Comment (grechaw) [inline marklogic-data-hub/src/main/java/com/marklogic/hub/impl/EntityManagerImpl.java]: You can remove this part and accompanying function.  just stick with all entities.

File: marklogic-data-hub/src/main/java/com/marklogic/hub/impl/EntityManagerImpl.java [hunk 3]
```diff
@@ -193,6 +207,31 @@ private HubModuleManager getPropsMgr() {
         return entities;
     }
 
+    private List<String> getEntityNamesInDatabase() {
```
Comment (grechaw) [inline marklogic-data-hub/src/main/java/com/marklogic/hub/impl/EntityManagerImpl.java]: You'll want to remove this too -- we don't want to check database state to decide what to do with the filesystem

File: ml-data-hub-plugin/src/test/groovy/com/marklogic/gradle/task/BaseTest.groovy [hunk 4]
```diff
@@ -144,6 +148,19 @@ class BaseTest extends Specification {
         DocumentBuilder builder = factory.newDocumentBuilder()
         return builder.parse(new File("src/test/resources/" + resourceName).getAbsoluteFile())
     }
+	
+	protected InputStream getResourceStream(String resourceName) {
+		return BaseTest.class.getClassLoader().getResourceAsStream(resourceName)
+	}
+	
+	protected JSONObject getJsonResource(String fileLoc) {
```
Comment (grechaw) [inline ml-data-hub-plugin/src/test/groovy/com/marklogic/gradle/task/BaseTest.groovy]: use jackson's API rather tha JSONObject here.  The type is JsonNode and we already use it everywhere.

**Comment by rahulvudutala** [comment]
I have addressed the comments. But I have amended few changes and did a force push to the previous commit. The changes in the previous commit are not reflected here.

**Comment by grechaw** [comment]
@rahulvudutala can you please rebase to develop and re-submit.  This PR is a little messed up and I don't think the work is complete.

**Comment by aebadirad** [comment]
See #1477 for the cleaned up branch

**Pull request closed without merging**